src/Socket.cpp

```diff
@@ -34,57 +36,54 @@ void Socket::write(const std::string& buff) {
     write(buff.c_str(), buff.size());
 }
 
-void Socket::read(char* buff, size_t buffsize) {
-    while(buffsize > 0) {
-        ssize_t res = ::read(_fd, buff, buffsize);
-        if(res == 0) {
-            throw std::runtime_error("Socket::read() - client closed connection");
-        } else if(res < 0) {
-            if(errno == EINTR)
-                continue;
-            throw std::runtime_error("Socket::read() - read() failed");
-        }
+void Socket::read(char* buff, const size_t buffsize) {
+    while(_buffer.size() < buffsize)
+        bufferedRead();
 
-        buff += res;
-        buffsize -= res;
-    }
+    strncpy(buff, _buffer.c_str(), buffsize);
```

Comment:
Just to be clear - in the future we can implement a second protocol, a binary one. For this reason Socket::read() and Socket::write() should work with binary data. Socket class should do no assumption about who is using it and what exactly is in the socket.